### PR TITLE
[SecurityBundle] Remove dead `class_exists` checks

### PR DESCRIPTION
**Pull request by chalasr**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

Now checking for more of these on upper branches.

### CODE REVIEW

File: src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/LdapFactoryTrait.php [hunk 1]
```diff
@@ -35,10 +35,6 @@ public function getKey(): string
     public function createAuthenticator(ContainerBuilder $container, string $firewallName, array $config, string $userProviderId): string
     {
         $key = str_replace('-', '_', $this->getKey());
-        if (!class_exists(LdapAuthenticator::class)) {
-            throw new \LogicException(sprintf('The "%s" authenticator requires the "symfony/ldap" package version "5.1" or higher.', $key));
```
Comment (chalasr) [inline src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/LdapFactoryTrait.php:39]: https://github.com/symfony/symfony/blob/efd6f4431fab8105eb46f79025e68f40b9d3eb83/src/Symfony/Bundle/SecurityBundle/composer.json#L59

File: src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/LoginLinkFactory.php [hunk 2]
```diff
@@ -88,10 +88,6 @@ public function getKey(): string
 
     public function createAuthenticator(ContainerBuilder $container, string $firewallName, array $config, string $userProviderId): string
     {
-        if (!class_exists(LoginLinkHandler::class)) {
-            throw new \LogicException('Login login link requires symfony/security-http:^5.2.');
```
Comment (chalasr) [inline src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/LoginLinkFactory.php:92]: https://github.com/symfony/symfony/blob/efd6f4431fab8105eb46f79025e68f40b9d3eb83/src/Symfony/Bundle/SecurityBundle/composer.json#L32

File: src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/LoginThrottlingFactory.php [hunk 3]
```diff
@@ -67,10 +67,6 @@ public function addConfiguration(NodeDefinition $builder)
 
     public function createAuthenticator(ContainerBuilder $container, string $firewallName, array $config, string $userProviderId): array
     {
-        if (!class_exists(LoginThrottlingListener::class)) {
-            throw new \LogicException('Login throttling requires symfony/security-http:^5.2.');
```
Comment (chalasr) [inline src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/LoginThrottlingFactory.php:71]: https://github.com/symfony/symfony/blob/efd6f4431fab8105eb46f79025e68f40b9d3eb83/src/Symfony/Bundle/SecurityBundle/composer.json#L32

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @chalasr.

**Pull request merged**